### PR TITLE
Guides for Serilog and EFCore

### PR DESCRIPTION
**Pull request by dukesteen**
Hi everyone, this will be my first contribution to Discord.Net!

I have discussed with Rozen on discord about creating these two guides. Feel free to review and suggest changes.

### CODE REVIEW

File: docs/guides/other_libs/serilog.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+|Name|Link|
+|--|--|
+|`Serilog.Extensions.Logging`| [link](https://www.nuget.org/packages/Serilog.Extensions.Logging)|
+|`Serilog.Sinks.Console`| [link](https://www.nuget.org/packages/Serilog.Sinks.Console)|
+
+## Configuring Serilog
+
+Serilog will be configured at the top of your async Main method, it looks like this
+
+```cs
```
Comment (csmir) [inline docs/guides/other_libs/serilog.md]: Sample into directory

File: docs/guides/other_libs/serilog.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    await Task.Delay(Timeout.Infinite);
+  }
+}
+```
+
+## Modifying your logging method
+
+For serilog to log discord events correctly, we have to map the discord `LogSeverity` to the serilog `LogEventLevel`. You can modify your log method to look like this.
+
+```cs
```
Comment (csmir) [inline docs/guides/other_libs/serilog.md]: Sample into directory

File: docs/guides/other_libs/serilog.md [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+## Testing
+
+If you run your application now, you should see something similar to this
+![Serilog output](images/serilog_output.png)
+
+## Using your new logger in other places
+
+Now that you have set up Serilog, you can use it everywhere in your application by simply calling
+
+```cs
```
Comment (csmir) [inline docs/guides/other_libs/serilog.md]: Sample into directory

File: docs/guides/other_libs/serilog.md [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+
+    _client.Log += LogAsync;
+
+    //  You can assign your bot token to a string, and pass that in to connect.
+    //  This is, however, insecure, particularly if you plan to have your code hosted in a public repository.
+    var token = "token";
+
+    // Some alternative options would be to keep your token in an Environment Variable or a standalone file.
+    // var token = Environment.GetEnvironmentVariable("NameOfYourEnvironmentVariable");
+    // var token = File.ReadAllText("token.txt")[0];
```
Comment (csmir) [inline docs/guides/other_libs/serilog.md]: My own mistake on initial review. ReadAllText returns string, ReadAllLines returns array. [0] is not required.

File: docs/guides/other_libs/serilog.md [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+Serilog will be configured at the top of your async Main method, it looks like this
+
+```cs
+using Discord;
+using Serilog;
+using Serilog.Events;
+
+public class Program
+{
+  public static Task Main(string[] args) => new Program().MainAsync().GetAwaiter().GetResult();
```
Comment (csmir) [inline docs/guides/other_libs/serilog.md]: May I suggest removing the `public` keyword, and making this `void`, so the entry point of this app is like any default console application format in standard2.1+?

File: docs/guides/other_libs/efcore.md [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+## Migrations
+
+Before you can start using your DbContext, you have to migrate the changes you've made in your code to your actual database.
+To learn more about migrations, visit the official Microsoft documentation [here](https://docs.microsoft.com/en-us/ef/core/managing-schemas/migrations/?tabs=dotnet-core-cli)
+
+## Using the DbContext
+
+You can now use the DbContext wherever you can inject it. Here's an example on injecting it into an interaction command module.
+
+```cs
```
Comment (csmir) [inline docs/guides/other_libs/efcore.md]: Sample into directory

File: docs/guides/other_libs/efcore.md [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+```
+
+> [!NOTE]
+> To learn more about creating the EFCore model, visit the following [link](https://docs.microsoft.com/en-us/ef/core/get-started/overview/first-app?tabs=netcore-cli#create-the-model)
+
+## Adding the DbContext to your Dependency Injection container
+
+To add your newly created DbContext to your Dependency Injection container, simply use the extension method provided by EFCore to add the context to your container. It should look something like this
+
+```cs
```
Comment (csmir) [inline docs/guides/other_libs/efcore.md]: Sample into directory

File: docs/guides/other_libs/efcore.md [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+|Name|Link|
+|--|--|
+| `Microsoft.EntityFrameworkCore` | [link](https://www.nuget.org/packages/Microsoft.EntityFrameworkCore) |
+| `Npgsql.EntityFrameworkCore.PostgreSQL` | [link](https://www.nuget.org/packages/Npgsql.EntityFrameworkCore.PostgreSQL)|
+
+## Configuring the DbContext
+
+To use EFCore, you need a DbContext to access everything in your database. The DbContext will look like this. Here is an example entity to show you how you can add more entities yourself later on.
+
+```cs
```
Comment (csmir) [inline docs/guides/other_libs/efcore.md]: Samples should go into /samples and referenced to as `[!code-csharp[title](samples/sample.cs)].`

File: docs/guides/other_libs/serilog.md [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+
+    // Block this task until the program is closed.
+    await Task.Delay(Timeout.Infinite);
+  }
+}
+```
+
+## Modifying your logging method
+
+For serilog to log discord events correctly, we have to map the discord `LogSeverity` to the serilog `LogEventLevel`. You can modify your log method to look like this.
```
Comment (csmir) [inline docs/guides/other_libs/serilog.md]: Discord*, also suggest changing `we` for `you`

**Review by csmir — CHANGES_REQUESTED** [review]

**Comment by dukesteen** [comment]
Fixed all review points in [acd62f8](https://github.com/discord-net/Discord.Net/pull/2134/commits/acd62f84473ed93be931414b781f793b3f236628)

File: docs/guides/other_libs/serilog.md [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+
+## Configuring Serilog
+
+Serilog will be configured at the top of your async Main method, it looks like this
+
+[!code-csharp[Configuring serilog](samples/ConfiguringSerilog.cs)]
+
+## Modifying your logging method
+
+For serilog to log Discord events correctly, we have to map the discord `LogSeverity` to the serilog `LogEventLevel`. You can modify your log method to look like this.
```
Comment (csmir) [inline docs/guides/other_libs/serilog.md]: Discord, Serilog*

File: docs/guides/other_libs/samples/ConfiguringSerilog.cs [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+
+        _client.Log += LogAsync;
+
+        //  You can assign your bot token to a string, and pass that in to connect.
+        //  This is, however, insecure, particularly if you plan to have your code hosted in a public repository.
+        var token = "token";
+
+        // Some alternative options would be to keep your token in an Environment Variable or a standalone file.
+        // var token = Environment.GetEnvironmentVariable("NameOfYourEnvironmentVariable");
+        // var token = File.ReadAllText("token.txt")[0];
```
Comment (csmir) [inline docs/guides/other_libs/samples/ConfiguringSerilog.cs]: Remove the `[0]`

**Review by csmir — CHANGES_REQUESTED** [review]
Addressing the same grammatical issues as before, including another.

**Comment by dukesteen** [comment]
Fixed in [502eea8](https://github.com/discord-net/Discord.Net/pull/2134/commits/502eea898d8413deb6420800397de5c24c07198c)

**Comment by csmir** [comment]
@quinchs this can be merged

**Pull request merged**